build/build/src/engine/context.rs

```diff
@@ -274,6 +274,9 @@ impl RunContext {
         // sbt:warning: java.lang.ClassNotFoundException:
         // org.enso.interpreter.node.expression.builtin.bool.True
         ide_ci::fs::remove_if_exists(&self.paths.repo_root.engine.runtime.target)?;
+        // cleanup distribution from previous build
+        // it is fast to assemble it again
+        ide_ci::fs::remove_if_exists(&self.paths.repo_root.built_distribution)?;
```

Comment:
Fix exit code `126` - [cannot execute binary file](https://github.com/enso-org/enso/actions/runs/10003911884/job/27651628322?pr=10602#step:7:984) by removing `built-distribution` debris before starting the build. 

How does the problem happen? 
When I configured `engine-runner/buildNative` to override output of buildEngineDistribution in
- #10374

I knew I was asking for problems. The _"override"_ only happens in certain CI jobs and only at the end of their execution. However, as previous output often stays around for next executor, we may face a _"debris problem"_ - the binary `enso` stays where `bash` `enso` script is expected.

Cleaning the `build-distribution` directory every time shall fix the problem.